py-polars/polars/internals/anonymous_scan.py

```diff
@@ -1,19 +1,14 @@
 from __future__ import annotations
 
-from functools import partial
-from typing import TYPE_CHECKING, Any, cast
+from typing import TYPE_CHECKING, Any
 
-import polars as pl
-from polars import internals as pli
 from polars.dependencies import pickle
-from polars.dependencies import pyarrow as pa  # noqa: TCH001
 
 if TYPE_CHECKING:
-    from polars.dataframe.frame import DataFrame
-    from polars.lazyframe.frame import LazyFrame
+    from polars import DataFrame
```

Comment:
How do we go about these imports, use `polars`, `polars.dataframe` or `polars.dataframe.frame`? I personally prefer the second one, i.e. we organize the `polars` module into a number of sub modules, and how those are organized is up to those modules, but that is maybe a bit arbitrary? Not sure I like the top-level import as that only works for the public api.

Reply:
Second one is fine! I'll adjust this PR accordingly.